src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -275,16 +280,30 @@ public Void run() throws InitializationException {
         }
     }
 
-    private AbstractReloadingMetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
+    private MetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
             throws Exception {
-        final AbstractReloadingMetadataResolver metadataResolver;
+        final AbstractMetadataResolver metadataResolver;
 
+        final String idpMetadataUrl = settings.get("idp.metadata_url");
+        final String idpMetadataFile = settings.get("idp.metadata_file");
+        final String idpMetadataBody = settings.get("idp.metadata_body");
         if (idpMetadataUrl != null) {
-            metadataResolver = new SamlHTTPMetadataResolver(settings, configPath);
+            metadataResolver = new SamlHTTPMetadataResolver(idpMetadataUrl, settings, configPath);
+        } else if (idpMetadataFile != null) {
+            metadataResolver = new SamlFilesystemMetadataResolver(idpMetadataFile, settings, configPath);
+        } else if (idpMetadataBody != null) {
+            metadataResolver = new DOMMetadataResolver(getMetadataDOM(idpMetadataBody));
         } else {
-            metadataResolver = new SamlFilesystemMetadataResolver(settings, configPath);
+            throw new Exception("One of idp.metadata_url, idp.metadata_file or idp.metadata_body must be configured");
         }
 
+        metadataResolver.setId(HTTPSamlAuthenticator.class.getName() + "_" + (++resolverIdCounter));
```

Comment:
What is the purpose of resolverIdCounter ? If the intent is creating a new id we can use uuid ?

Reply:
I based this of the previous implementation - https://github.com/opendistro-for-elasticsearch/security/blob/167acc0769397b765fa52fd68a4d555524915deb/src/main/java/com/amazon/dlic/auth/http/saml/SamlHTTPMetadataResolver.java#L40. 

Any reason for preferring UUID?

Reply:
Just curious about the role of resolverIdCounter. 
If this id is passed as output or having a distributed nature from different nodes then uuid is preferred to avoid conflicts.

Reply:
My guess is that `id` is used in logging only to identify the resolver, so usage of uuid will be less preferable over the counter.

Reply:
Okay. I assumed `setId` would be used for other purposes. If it's only used in logging, don't know how useful the number would be to identify it. 

I see its the current implementation

Reply:
I guess instead of a counter we may add `id` spec as an optional field to the SAML authenticator setting to be provided by an admin, so admin can more easily identify  authenticator metadata resolver.